lib/wallet/http.js

```diff
@@ -1074,12 +1074,20 @@ class HTTP extends Server {
       const passphrase = valid.str('passphrase');
       const broadcast = valid.bool('broadcast', true);
       const sign = valid.bool('sign', true);
+      const idempotencyKey = valid.str('idempotencyKey');
 
       assert(name, 'Name is required.');
       assert(broadcast ? sign : true, 'Must sign when broadcasting.');
 
+      if (idempotencyKey) {
+        assert(broadcast && sign, 'Must sign and broadcast if using idempotency cache')
+      }
+
       const options = TransactionOptions.fromValidator(valid);
-      const mtx = await req.wallet.createOpen(name, force, options);
+      const [mtx, fromCache] = await req.wallet.withOpenCache(idempotencyKey, () =>
```

Comment:
style nit: implicit returns should only be used for single line functions, too easy to miss while skimming, use
```javascript
...withOpenCache(idempotencyKey, () => {
  return req.wallet.createOpen(name, force, options);
});
```